core/src/error.rs

```diff
@@ -1,8 +1,38 @@
 //! `Ipld` error definitions.
 use crate::ipld::{Ipld, IpldIndex};
-use std::convert::TryFrom;
+pub use anyhow::{Error, Result};
```

Comment:
I think it would. Using enums for errors seems like an anti pattern to me. After using error enums and things getting messy I started using this pattern in all my codebases. It should also be faster according to @withoutboats, because the happy path isn't penalized. Most errors are fatal errors not handled but only propagated. In the case that an error needs to be handled by an application downcast_ref works great.